tests/collector/config_parser/config_parser.cpp

```diff
@@ -0,0 +1,250 @@
+   modify it under the terms of the GNU Lesser General Public
+   License as published by the Free Software Foundation; either
+   version 3.0 of the License, or (at your option) any later version.
+
+   This library is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Lesser General Public License for more details.
+
+   You should have received a copy of the GNU Lesser General Public
+   License along with this library.
+*/
+
+#include <Config.h>
+#include <ConfigParser.h>
+
+#include <rapidjson/writer.h>
+
+#include <gtest/gtest.h>
+
+namespace {
+
+// Fill up a test object with the values which are known to be different from defaults.
+struct TestConfig : public Config
+{
+    TestConfig()
+    {
+        monitor_port = Config::Default::monitor_port + 1;
+        wait_timeout = Config::Default::wait_timeout + 2;
+        forbidden_dht_groups = !!Config::Default::forbidden_dht_groups;
+        forbidden_unmatched_group_total_space = !!Config::Default::forbidden_unmatched_group_total_space;
+        forbidden_ns_without_settings = !!Config::Default::forbidden_ns_without_settings;
+        forbidden_dc_sharing_among_groups = !!Config::Default::forbidden_dc_sharing_among_groups;
+        reserved_space = Config::Default::reserved_space + 7;
+        node_backend_stat_stale_timeout = Config::Default::node_backend_stat_stale_timeout + 8;
+        dnet_log_mask = Config::Default::dnet_log_mask + 9;
+        net_thread_num = Config::Default::net_thread_num + 10;
+        io_thread_num = Config::Default::io_thread_num + 11;
+        nonblocking_io_thread_num = Config::Default::nonblocking_io_thread_num + 12;
+        infrastructure_dc_cache_update_period = Config::Default::infrastructure_dc_cache_update_period + 13;
+        infrastructure_dc_cache_valid_time = Config::Default::infrastructure_dc_cache_valid_time + 14;
+        inventory_worker_timeout = Config::Default::inventory_worker_timeout + 15;
+
+        app_name = "mastermind2.26";
+        cache_group_path_prefix = "/cache/group/path/prefix";
+
+        NodeInfo info;
+        info.host = "node1.example.com";
+        info.port = 10017;
+        info.family = 2;
+        nodes.push_back(info);
+        info.host = "node2.example.com";
+        info.port = 10023;
+        info.family = 10;
+        nodes.push_back(info);
+
+        metadata.url = "mongodb://mongo1.example.com:27017,mongo2.example.com:27017/?replicaSet=test-mongodb";
+        metadata.options.connectTimeoutMS = Default::metadata_options_connectTimeoutMS + 16;
+        metadata.history.db = "history_db";
+        metadata.inventory.db = "inventory_db";
+        metadata.jobs.db = "jobs_db";
+    }
+};
+
+TestConfig s_config;
+
+// Generate JSON config.
+std::string generate_json()
+{
+    rapidjson::StringBuffer buf;
+    rapidjson::Writer<rapidjson::StringBuffer> writer(buf);
```

Comment:
I supposed that we agreed that you will use raw string literals with final json string, weren't we?